com.unity.ml-agents.extensions/Runtime/Sensors/GridSensor.cs

```diff
@@ -98,6 +99,20 @@ public enum GridDepthType { Channel, ChannelHot };
         [Tooltip("The reference of the root of the agent. This is used to disambiguate objects with the same tag as the agent. Defaults to current GameObject")]
         public GameObject rootReference;
 
+        [Header("Collider Buffer Properties")]
+        [Tooltip("The absolute max size of the Collider buffer used in the non-allocating Physics calls.  In other words" +
+            " the Collider buffer will never grow beyond this number even if there are more Colliders in the Grid Cell.")]
+        public int AbsoluteMaxCollidersPerCell = 500;
+        [Tooltip(
+            "The Estimated Max Number of Colliders to expect per cell.  This number is used to " +
+            "pre-allocate an array of Colliders in order to take advantage of the OverlapBoxNonAlloc " +
+            "Physics API.  If the number of colliders found is >= EstimatedMaxCollidersPerCell the array " +
+            "will be resized to double its current size.  The hard coded absolute size is 500.")]
+        public int EstimatedMaxCollidersPerCell = 4;
```

Comment:
Maybe InitialColliderBufferSize and MaxColliderBufferSize? I think the `PerCell` is redundant and maybe even a bit misleading, because it could sound like we'd make the buffer (4 * number of cells) big.